File: abelkhan_type_csharp_component_msgpack/component/dbproxy/centerproxy.cs
/*
 * centerproxy
 * 2020/6/3
 * qianqians
 */

namespace abelkhan
{
	public class centerproxy
	{
		private center_caller _caller;
		public bool is_reg_sucess;
		public centerproxy(abelkhan.Ichannel ch, modulemng modules)
		{
			is_reg_sucess = false;
			_caller = new center_caller(ch, modules);
		}

		public void reg_dbproxy(string svr_name, string ip, ushort port)
		{
			_caller.reg_server("dbproxy", "dbproxy", svr_name, ip, port);
		}
	}
}

File: abelkhan_type_component/component/dbproxy/centerproxy.ts
import abelkhan = require('../../abelkhan_type/ts/abelkhan');
import center = require('../../protcol/ts/center');

export class centerproxy{
	private _center_caller : center.center_caller;
	public is_reg_sucess : boolean;
	constructor(ch:abelkhan.Ichannel, _modules:abelkhan.modulemng){
		this._center_caller = new center.center_caller(ch, _modules);
		this.is_reg_sucess = false;
	}

	public reg_dbproxy(svr_name:string, ip:string, port:number){
		this._center_caller.reg_server("dbproxy", svr_name, ip, port);
	}

}
